unlock_indicator.c

```diff
@@ -280,31 +316,16 @@ void draw_image(xcb_pixmap_t bg_pixmap, uint32_t *resolution) {
         }
 
         if (text) {
-            cairo_text_extents_t extents;
-            double x, y;
-
-            cairo_text_extents(ctx, text, &extents);
-            x = BUTTON_CENTER - ((extents.width / 2) + extents.x_bearing);
-            y = BUTTON_CENTER - ((extents.height / 2) + extents.y_bearing);
-
-            cairo_move_to(ctx, x, y);
-            cairo_show_text(ctx, text);
-            cairo_close_path(ctx);
+            display_button_text(ctx, text, 0.);
         }
 
         if (modifier_string != NULL) {
-            cairo_text_extents_t extents;
-            double x, y;
-
             cairo_set_font_size(ctx, 14.0);
-
-            cairo_text_extents(ctx, modifier_string, &extents);
-            x = BUTTON_CENTER - ((extents.width / 2) + extents.x_bearing);
-            y = BUTTON_CENTER - ((extents.height / 2) + extents.y_bearing) + 28.0;
-
-            cairo_move_to(ctx, x, y);
-            cairo_show_text(ctx, modifier_string);
-            cairo_close_path(ctx);
+            display_button_text(ctx, modifier_string, 28.);
+        }
+        if (layout_string != NULL) {
+            cairo_set_font_size(ctx, 14.0);
+            display_button_text(ctx, layout_string, 56.);
```

Comment:
Instead of displaying the layout below the modifier text, can you try displaying it above the `text`?

Reply:
done